src/ecdsa.js

```diff
@@ -10,12 +10,12 @@ var ONE = new Buffer([1])
 
 // https://tools.ietf.org/html/rfc6979#section-3.2
 function deterministicGenerateK (curve, hash, d, checkSig) {
-  typeForce('Buffer', hash)
-  typeForce('BigInteger', d)
-  typeForce('Function', checkSig)
-
-  // sanity check
-  assert.equal(hash.length, 32, 'Hash must be 256 bit')
+  typeforce(types.tuple(
+    types.ECCurve,
+    types.Hash256bit,
+    types.BigInt,
+    types.Function
+  ), arguments)
```

Comment:
All these types are currently being created on the fly (though, it is pretty light weight), we can pull them out of the function and just enforce them on the first line if we feel performance is an issue,  though,  this new syntax is already 1000% faster than the previous implementation.